### ISSUE TITLE
`Wrong-type-argument: char-or-string-p nil` when using `org-roam-insert-immediate` and when capturing the buffer in `org-roam-insert`

### ISSUE THREAD

**Issue by ln0a** [org-roam/org-roam#1075]
Vanilla Emacs 27.1.50 OSX

When using `org-roam-insert` (`-immediate`) I encounter a `Wrong-type-argument: char-or-string-p nil` error. The new file is created and saved fine, but when using `org-roam-insert-immediate` on marked text the link is not applied and the command must be re-run. 

#### Backtrace

```
Debugger entered--Lisp error: (wrong-type-argument char-or-string-p nil)
  ucs-normalize-NFD-string(nil)
  #f(compiled-function (--cl-nonspacing-mark-p-- s) #<bytecode 0x1fe46a1d3cd1>)(#f(compiled-function (char) #<bytecode 0x1fe46a1d3cb9>) nil)
  org-roam--title-to-slug(nil)
  org-roam--update-file-name-on-title-change(nil "test")
  run-hook-with-args(org-roam--update-file-name-on-title-change nil "test")
  org-roam--handle-title-change()
  run-hooks(after-save-hook)
  basic-save-buffer(nil)
  save-buffer()
  org-roam-capture--save-file-maybe()
  org-roam-capture--finalize()
  run-hooks(org-capture-after-finalize-hook)
  (let ((abort-note nil)) (org-capture-put :captured-entry-size (- (point-max) (point-min))) (widen) (org-capture-put :insertion-point (point)) (if org-note-abort (let ((beg (org-capture-get :begin-marker 'local)) (end (org-capture-get :end-marker 'local))) (if (not (and beg end)) (setq abort-note 'dirty) (setq abort-note t) (save-excursion (save-restriction (widen) (kill-region beg end))))) (if (derived-mode-p 'org-mode) (progn (save-excursion (if (condition-case nil (progn ...) (error nil)) (progn (org-update-parent-todo-statistics) (org-update-checkbox-count)))) (if (and (eq (org-capture-get :type ...) 'table-line) (org-at-table-p)) (progn (if (not ...) (org-table-align) (if ... ...) (org-table-recalculate ...)))))) (org-capture-store-last-position) (run-hooks 'org-capture-before-finalize-hook)) (if (org-capture-get :decrypted) (progn (save-excursion (goto-char (org-capture-get :decrypted)) (org-encrypt-entry)))) (if (org-capture-get :no-save) nil (save-buffer)) (let ((return-wconf (org-capture-get :return-to-wconf 'local)) (new-buffer (org-capture-get :new-buffer 'local)) (kill-buffer (org-capture-get :kill-buffer 'local)) (base-buffer (buffer-base-buffer (current-buffer)))) (kill-buffer (current-buffer)) (save-current-buffer (set-buffer (org-capture-get :buffer)) (let ((reg (org-capture-get :initial-target-region)) (pos (org-capture-get :initial-target-position)) (ipt (org-capture-get :insertion-point)) (size (org-capture-get :captured-entry-size))) (if (not reg) (widen) (cond ((< ipt ...) (narrow-to-region ... ...)) ((> ipt ...) (narrow-to-region ... ...)) (t (narrow-to-region ... ...))) (if (< ipt (car reg)) (goto-char (+ size pos)) (goto-char (if ... ... pos)))))) (if (and base-buffer new-buffer kill-buffer) (progn (save-current-buffer (set-buffer base-buffer) (save-buffer)) (kill-buffer base-buffer))) (set-window-configuration return-wconf)) (run-hooks 'org-capture-after-finalize-hook) (cond (abort-note (cond ((eql abort-note 'clean) (message "Capture process aborted and target buffer cleaned ...")) ((eql abort-note 'dirty) (error "Capture process aborted, but target buffer could n...")))) (stay-with-capture (org-capture-goto-last-stored))) (not abort-note))
  (closure (org-remember-templates org-remember-default-headline org-end-time-was-given t) (&optional stay-with-capture) "Finalize the capture process.\nWith prefix argument..." (interactive "P") (if (org-capture-get :jump-to-captured) (progn (setq stay-with-capture t))) (if (and org-capture-mode (buffer-base-buffer (current-buffer))) nil (error "This does not seem to be a capture buffer for Org ...")) (run-hooks 'org-capture-prepare-finalize-hook) (if (and org-capture-clock-was-started org-clock-marker (eq (marker-buffer org-clock-marker) (buffer-base-buffer)) (>= org-clock-marker (point-min)) (< org-clock-marker (point-max))) (progn (if org-capture-clock-keep (org-clock-update-mode-line t) (let (org-log-note-clock-out) (org-clock-out)) (if (and (org-capture-get :clock-resume 'local) (markerp (org-capture-get :interrupted-clock ...)) (buffer-live-p (marker-buffer ...))) (progn (let (...) (let ... ...)) (message "Interrupted clock has been resumed")))))) (let ((abort-note nil)) (org-capture-put :captured-entry-size (- (point-max) (point-min))) (widen) (org-capture-put :insertion-point (point)) (if org-note-abort (let ((beg (org-capture-get :begin-marker 'local)) (end (org-capture-get :end-marker 'local))) (if (not (and beg end)) (setq abort-note 'dirty) (setq abort-note t) (save-excursion (save-restriction (widen) (kill-region beg end))))) (if (derived-mode-p 'org-mode) (progn (save-excursion (if (condition-case nil ... ...) (progn ... ...))) (if (and (eq ... ...) (org-at-table-p)) (progn (if ... ... ... ...))))) (org-capture-store-last-position) (run-hooks 'org-capture-before-finalize-hook)) (if (org-capture-get :decrypted) (progn (save-excursion (goto-char (org-capture-get :decrypted)) (org-encrypt-entry)))) (if (org-capture-get :no-save) nil (save-buffer)) (let ((return-wconf (org-capture-get :return-to-wconf 'local)) (new-buffer (org-capture-get :new-buffer 'local)) (kill-buffer (org-capture-get :kill-buffer 'local)) (base-buffer (buffer-base-buffer (current-buffer)))) (kill-buffer (current-buffer)) (save-current-buffer (set-buffer (org-capture-get :buffer)) (let ((reg (org-capture-get :initial-target-region)) (pos (org-capture-get :initial-target-position)) (ipt (org-capture-get :insertion-point)) (size (org-capture-get :captured-entry-size))) (if (not reg) (widen) (cond (... ...) (... ...) (t ...)) (if (< ipt ...) (goto-char ...) (goto-char ...))))) (if (and base-buffer new-buffer kill-buffer) (progn (save-current-buffer (set-buffer base-buffer) (save-buffer)) (kill-buffer base-buffer))) (set-window-configuration return-wconf)) (run-hooks 'org-capture-after-finalize-hook) (cond (abort-note (cond ((eql abort-note 'clean) (message "Capture process aborted and target buffer cleaned ...")) ((eql abort-note 'dirty) (error "Capture process aborted, but target buffer could n...")))) (stay-with-capture (org-capture-goto-last-stored))) (not abort-note)))()
  apply((closure (org-remember-templates org-remember-default-headline org-end-time-was-given t) (&optional stay-with-capture) "Finalize the capture process.\nWith prefix argument..." (interactive "P") (if (org-capture-get :jump-to-captured) (progn (setq stay-with-capture t))) (if (and org-capture-mode (buffer-base-buffer (current-buffer))) nil (error "This does not seem to be a capture buffer for Org ...")) (run-hooks 'org-capture-prepare-finalize-hook) (if (and org-capture-clock-was-started org-clock-marker (eq (marker-buffer org-clock-marker) (buffer-base-buffer)) (>= org-clock-marker (point-min)) (< org-clock-marker (point-max))) (progn (if org-capture-clock-keep (org-clock-update-mode-line t) (let (org-log-note-clock-out) (org-clock-out)) (if (and (org-capture-get :clock-resume ...) (markerp ...) (buffer-live-p ...)) (progn (let ... ...) (message "Interrupted clock has been resumed")))))) (let ((abort-note nil)) (org-capture-put :captured-entry-size (- (point-max) (point-min))) (widen) (org-capture-put :insertion-point (point)) (if org-note-abort (let ((beg (org-capture-get :begin-marker ...)) (end (org-capture-get :end-marker ...))) (if (not (and beg end)) (setq abort-note 'dirty) (setq abort-note t) (save-excursion (save-restriction ... ...)))) (if (derived-mode-p 'org-mode) (progn (save-excursion (if ... ...)) (if (and ... ...) (progn ...)))) (org-capture-store-last-position) (run-hooks 'org-capture-before-finalize-hook)) (if (org-capture-get :decrypted) (progn (save-excursion (goto-char (org-capture-get :decrypted)) (org-encrypt-entry)))) (if (org-capture-get :no-save) nil (save-buffer)) (let ((return-wconf (org-capture-get :return-to-wconf 'local)) (new-buffer (org-capture-get :new-buffer 'local)) (kill-buffer (org-capture-get :kill-buffer 'local)) (base-buffer (buffer-base-buffer (current-buffer)))) (kill-buffer (current-buffer)) (save-current-buffer (set-buffer (org-capture-get :buffer)) (let ((reg ...) (pos ...) (ipt ...) (size ...)) (if (not reg) (widen) (cond ... ... ...) (if ... ... ...)))) (if (and base-buffer new-buffer kill-buffer) (progn (save-current-buffer (set-buffer base-buffer) (save-buffer)) (kill-buffer base-buffer))) (set-window-configuration return-wconf)) (run-hooks 'org-capture-after-finalize-hook) (cond (abort-note (cond ((eql abort-note ...) (message "Capture process aborted and target buffer cleaned ...")) ((eql abort-note ...) (error "Capture process aborted, but target buffer could n...")))) (stay-with-capture (org-capture-goto-last-stored))) (not abort-note))) nil)
  org-capture-finalize()
  (progn (org-capture-finalize))
  (if (org-capture-get :immediate-finish) (progn (org-capture-finalize)))
  (cond ((equal entry "C") (customize-variable 'org-capture-templates)) ((equal entry "q") (user-error "Abort")) (t (org-capture-set-plist entry) (org-capture-get-template) (org-capture-put :original-buffer orig-buf :original-file (or (buffer-file-name orig-buf) (and (featurep 'dired) (car (rassq orig-buf dired-buffers)))) :original-file-nondirectory (and (buffer-file-name orig-buf) (file-name-nondirectory (buffer-file-name orig-buf))) :annotation annotation :initial initial :return-to-wconf (current-window-configuration) :default-time (or org-overriding-default-time (org-current-time))) (org-capture-set-target-location (and (equal goto 0) 'here)) (condition-case error (org-capture-put :template (org-capture-fill-template)) ((error quit) (if (get-buffer "*Capture*") (kill-buffer "*Capture*")) (error "Capture abort: %s" (error-message-string error)))) (setq org-capture-clock-keep (org-capture-get :clock-keep)) (condition-case error (org-capture-place-template (eq (car (org-capture-get :target)) 'function)) ((error quit) (if (and (buffer-base-buffer (current-buffer)) (string-prefix-p "CAPTURE-" (buffer-name))) (progn (kill-buffer (current-buffer)))) (set-window-configuration (org-capture-get :return-to-wconf)) (error "Capture template `%s': %s" (org-capture-get :key) (error-message-string error)))) (if (and (derived-mode-p 'org-mode) (org-capture-get :clock-in)) (progn (condition-case nil (progn (if (org-clock-is-active) (progn ...)) (org-clock-in) (set (make-local-variable ...) t)) (error "Could not start the clock in this capture buffer")))) (if (org-capture-get :immediate-finish) (progn (org-capture-finalize)))))
  (let* ((orig-buf (current-buffer)) (annotation (if (and (boundp 'org-capture-link-is-already-stored) org-capture-link-is-already-stored) (plist-get org-store-link-plist :annotation) (condition-case nil (progn (org-store-link nil)) (error nil)))) (entry (or org-capture-entry (org-capture-select-template keys))) initial) (setq initial (or org-capture-initial (and (org-region-active-p) (buffer-substring (point) (mark))))) (if (stringp initial) (progn (remove-text-properties 0 (length initial) '(read-only t) initial))) (if (stringp annotation) (progn (remove-text-properties 0 (length annotation) '(read-only t) annotation))) (cond ((equal entry "C") (customize-variable 'org-capture-templates)) ((equal entry "q") (user-error "Abort")) (t (org-capture-set-plist entry) (org-capture-get-template) (org-capture-put :original-buffer orig-buf :original-file (or (buffer-file-name orig-buf) (and (featurep 'dired) (car (rassq orig-buf dired-buffers)))) :original-file-nondirectory (and (buffer-file-name orig-buf) (file-name-nondirectory (buffer-file-name orig-buf))) :annotation annotation :initial initial :return-to-wconf (current-window-configuration) :default-time (or org-overriding-default-time (org-current-time))) (org-capture-set-target-location (and (equal goto 0) 'here)) (condition-case error (org-capture-put :template (org-capture-fill-template)) ((error quit) (if (get-buffer "*Capture*") (kill-buffer "*Capture*")) (error "Capture abort: %s" (error-message-string error)))) (setq org-capture-clock-keep (org-capture-get :clock-keep)) (condition-case error (org-capture-place-template (eq (car (org-capture-get :target)) 'function)) ((error quit) (if (and (buffer-base-buffer ...) (string-prefix-p "CAPTURE-" ...)) (progn (kill-buffer ...))) (set-window-configuration (org-capture-get :return-to-wconf)) (error "Capture template `%s': %s" (org-capture-get :key) (error-message-string error)))) (if (and (derived-mode-p 'org-mode) (org-capture-get :clock-in)) (progn (condition-case nil (progn (if ... ...) (org-clock-in) (set ... t)) (error "Could not start the clock in this capture buffer")))) (if (org-capture-get :immediate-finish) (progn (org-capture-finalize))))))
  (cond ((equal goto '(4)) (org-capture-goto-target)) ((equal goto '(16)) (org-capture-goto-last-stored)) (t (let* ((orig-buf (current-buffer)) (annotation (if (and (boundp ...) org-capture-link-is-already-stored) (plist-get org-store-link-plist :annotation) (condition-case nil (progn ...) (error nil)))) (entry (or org-capture-entry (org-capture-select-template keys))) initial) (setq initial (or org-capture-initial (and (org-region-active-p) (buffer-substring (point) (mark))))) (if (stringp initial) (progn (remove-text-properties 0 (length initial) '(read-only t) initial))) (if (stringp annotation) (progn (remove-text-properties 0 (length annotation) '(read-only t) annotation))) (cond ((equal entry "C") (customize-variable 'org-capture-templates)) ((equal entry "q") (user-error "Abort")) (t (org-capture-set-plist entry) (org-capture-get-template) (org-capture-put :original-buffer orig-buf :original-file (or (buffer-file-name orig-buf) (and ... ...)) :original-file-nondirectory (and (buffer-file-name orig-buf) (file-name-nondirectory ...)) :annotation annotation :initial initial :return-to-wconf (current-window-configuration) :default-time (or org-overriding-default-time (org-current-time))) (org-capture-set-target-location (and (equal goto 0) 'here)) (condition-case error (org-capture-put :template (org-capture-fill-template)) ((error quit) (if ... ...) (error "Capture abort: %s" ...))) (setq org-capture-clock-keep (org-capture-get :clock-keep)) (condition-case error (org-capture-place-template (eq ... ...)) ((error quit) (if ... ...) (set-window-configuration ...) (error "Capture template `%s': %s" ... ...))) (if (and (derived-mode-p ...) (org-capture-get :clock-in)) (progn (condition-case nil ... ...))) (if (org-capture-get :immediate-finish) (progn (org-capture-finalize))))))))
  org-capture(nil "d")
  org-roam-capture--capture()
  org-roam-insert(nil)
  apply(org-roam-insert nil)
  org-roam-insert-immediate(nil)
  funcall-interactively(org-roam-insert-immediate nil)
  call-interactively(org-roam-insert-immediate record nil)
  command-execute(org-roam-insert-immediate record)
```

**Comment by jethrokuan** [comment]
Fixed by #1076

**Comment by ln0a** [comment]
Apreciate it, thanks man
